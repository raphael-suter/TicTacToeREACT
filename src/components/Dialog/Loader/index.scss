@import "../../../styles/colors.scss";

$size: 50px;
$thickness: 0.5rem;

.loader {
  width: $size;
  height: $size;
  margin: 0.6rem auto;
  border: $thickness solid $mainColor;
  border-top: $thickness solid $mediumColor;
  border-radius: 50%;
  animation: dialog-loader-spin 2s linear infinite;
}

@keyframes dialog-loader-spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
