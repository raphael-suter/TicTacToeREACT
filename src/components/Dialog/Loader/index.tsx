import React from 'react';
import styled, { keyframes } from 'styled-components';

const size = '50px';
const thickness = '0.5rem';

const spin = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const Loader = styled.div`
  width: ${size};
  height: ${size};
  margin: 0.6rem auto;
  border: ${thickness} solid ${({ theme }) => theme.mainColor};
  border-top: ${thickness} solid ${({ theme }) => theme.mediumColor};
  border-radius: 50%;
  animation: ${spin} 2s linear infinite;
`;

export default () => <Loader />;
