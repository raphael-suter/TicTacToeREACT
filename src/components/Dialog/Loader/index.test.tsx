import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Loader from '.';

describe('Loader', () => {
  test('Renders correctly.', () => {
    const wrapper = shallow(<Loader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
