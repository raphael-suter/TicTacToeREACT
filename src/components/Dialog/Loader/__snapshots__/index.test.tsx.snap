// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Renders correctly. 1`] = `<styled.div />`;
